apps/web-app/components/convos/convoListItem.vue

```diff
@@ -11,128 +12,103 @@
   type Props = {
     convo: UserConvosDataType[number];
   };
-
   const props = defineProps<Props>();
 
   const timeAgo = useTimeAgo(props.convo.lastUpdatedAt || new Date());
-  const authorName = computed(() => {
-    if (
-      props.convo.messages[0].author.userProfile?.firstName &&
-      props.convo.messages[0].author.userProfile?.lastName
-    ) {
-      return (
-        props.convo.messages[0].author.userProfile.firstName +
-        ' ' +
-        props.convo.messages[0].author.userProfile.lastName
-      );
-    }
-    if (props.convo.messages[0].author.userGroup?.name) {
-      return props.convo.messages[0].author.userGroup.name + ' Team';
-    }
 
-    if (props.convo.messages[0].author.foreignEmailIdentity?.senderName) {
-      return props.convo.messages[0].author.foreignEmailIdentity.senderName;
-    }
-    return '';
+  const participantArray = ref<ConvoParticipantEntry[]>([]);
+  const author = ref<ConvoParticipantEntry>();
+  const authorEntryPublicId = computed(() => {
+    const entryAuthor = props.convo.entries[0].author;
+    return (
+      entryAuthor.orgMember?.publicId ||
+      entryAuthor.userGroup?.publicId ||
+      entryAuthor.contact?.publicId
+    );
   });
-
-  const authorAvatar = computed(() => {
-    if (props.convo.messages[0].author.userProfile?.avatarId) {
-      return props.convo.messages[0].author.userProfile.avatarId;
-    }
-    if (props.convo.messages[0].author.userGroup?.avatarId) {
-      return props.convo.messages[0].author.userGroup.avatarId;
-    }
-
-    if (props.convo.messages[0].author.foreignEmailIdentity?.avatarId) {
-      return props.convo.messages[0].author.foreignEmailIdentity.avatarId;
-    }
-    return '';
+  const authorEntryName = computed(() => {
+    const entryAuthor = props.convo.entries[0].author;
+    return (
+      entryAuthor.userGroup?.publicId ||
+      entryAuthor.contact?.publicId ||
+      entryAuthor.orgMember?.profile.firstName +
+        ' ' +
+        entryAuthor.orgMember?.profile.lastName ||
+      'Participant'
+    );
   });
 
-  const authorColor = computed(() => {
-    if (props.convo.messages[0].author.userGroup?.color) {
-      return props.convo.messages[0].author.userGroup.color;
+  for (const participant of props.convo.participants) {
+    const participantPublicId = participant.publicId;
+    const participantTypePublicId =
+      participant.contact?.publicId ||
+      participant.userGroup?.publicId ||
+      participant.orgMember?.publicId ||
+      '';
+    const avatarPublicId =
+      participant.contact?.avatarId ||
+      participant.userGroup?.avatarId ||
+      participant.orgMember?.profile.avatarId ||
+      '';
```

Comment:
I feel like those should be converted to individual functions because they will be most like reused a lot